packages/react-app/src/components/giveaways/MainContent.tsx

```diff
@@ -1,13 +1,34 @@
 import { format } from 'date-fns';
-import { useContext } from 'react';
+import { useContext, useMemo } from 'react';
+import { useNavigate } from 'react-router-dom';
 
-import { Stack, Typography } from '@mui/material';
+import ErrorIcon from '@mui/icons-material/Error';
+import { Button, Stack, Typography } from '@mui/material';
 
-import { Giveaway } from '../../lib/types';
-import { GiveawayContext } from '../../pages/details';
+import useUserInfo from '../../hooks/useUserInfo';
+import { GiveawayContext, GiveawayDetailData } from '../../pages/details';
+import { UserRole } from '../../lib/types';
 
 const GiveawayMainContent = () => {
-  const giveaway = useContext(GiveawayContext) as Giveaway;
+  const navigate = useNavigate();
+  const userInfo = useUserInfo();
+  const isAdmin = userInfo?.role === UserRole.ADMIN;
+
+  const { giveaway, participants } = useContext(
+    GiveawayContext
+  ) as GiveawayDetailData;
+
+  const nrParticipants = useMemo(() => {
```

Comment:
There is a more efficient way to do this by adding a virtual property to the giveaway in the backend that will calculate the number of participants: https://mongoosejs.com/docs/tutorials/virtuals.html That way we reduce the number of calculations in the frontend since the client might have limited processing power, making the app really slow

Reply:
Since in this case the number of participants depends on the participants field, I believe it is not possible to return it in the list/detail endpoint views, participants data are in a separated endpoint